plugins/inputs/logstash/logstash.go

```diff
@@ -0,0 +1,291 @@
+  ## This plugin reads metrics exposed by Logstash Monitoring API.
+  #
+  # url = "http://localhost:9600"
+  url = "http://localhost:9600"
+`
+const jvmStats = "/_node/stats/jvm"
+const processStats = "/_node/stats/process"
+const pipelineStats = "/_node/stats/pipelines"
+
+type Logstash struct {
+	URL    string
+	client *http.Client
+}
+
+type JVMStats struct {
+	ID  string      `json:"id"`
+	JVM interface{} `json:"jvm"`
+}
+
+type ProcessStats struct {
+	ID      string      `json:"id"`
+	Process interface{} `json:"process"`
+}
+
+type PluginEvents struct {
+	QueuePushDurationInMillis float64 `json:"queue_push_duration_in_millis"`
+	DurationInMillis          float64 `json:"duration_in_millis"`
+	In                        float64 `json:"in"`
+	Out                       float64 `json:"out"`
+}
+
+type Plugin struct {
+	ID     string       `json:"id"`
+	Events PluginEvents `json:"events"`
+	Name   string       `json:"name"`
+}
+
+type PipelinePlugins struct {
+	Inputs  []Plugin `json:"inputs"`
+	Filters []Plugin `json:"filters"`
+	Outputs []Plugin `json:"outputs"`
+}
+
+type PipelineQueue struct {
+	Events   float64     `json:"events"`
+	Qtype    string      `json:"type"`
+	Capacity interface{} `json:"capacity"`
+	Data     interface{} `json:"data"`
+}
+
+type Pipeline struct {
+	Events  interface{}     `json:"events"`
+	Plugins PipelinePlugins `json:"plugins"`
+	Reloads interface{}     `json:"reloads"`
+	Queue   PipelineQueue   `json:"queue"`
+}
+
+type PipelineTopLevel struct {
+	Pipeline_Monitoring interface{} `json:".monitoring-logstash"`
+	Pipeline_Main       Pipeline    `json:"main"`
+}
+
+type PipelineStats struct {
+	ID        string           `json:"id"`
+	Pipelines PipelineTopLevel `json:"pipelines"`
+}
+
+//Description returns short info about plugin
+func (l *Logstash) Description() string { return "Read metrics exposed by Logstash" }
+
+//SampleConfig returns details how to configure plugin
+func (l *Logstash) SampleConfig() string { return sampleConfig }
+
+//createHttpClient create clients to access API
+func (l *Logstash) createHTTPClient() (*http.Client, error) {
+
+	client := &http.Client{
+		Transport: &http.Transport{
+			Proxy: http.ProxyFromEnvironment,
+		},
+		Timeout: time.Duration(4 * time.Second),
+	}
+
+	return client, nil
+}
+
+func (l *Logstash) gatherJSONData(url string, v interface{}) error {
+
+	r, err := l.client.Get(url)
+	if err != nil {
+		return err
+	}
+
+	defer r.Body.Close()
+
+	if r.StatusCode != http.StatusOK {
+		return fmt.Errorf("Logstash: API responded with status-code %d, expected %d",
+			r.StatusCode, http.StatusOK)
+	}
+
+	if err = json.NewDecoder(r.Body).Decode(v); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+func (l *Logstash) gatherJVMStats(url string, acc telegraf.Accumulator) error {
+	JVMStats := &JVMStats{}
+	if err := l.gatherJSONData(url, JVMStats); err != nil {
+		return err
+	}
+
+	tags := map[string]string{
+		"node_id": JVMStats.ID,
+	}
+
+	stats := map[string]interface{}{
```

Comment:
rather than allocating a `stats` map, why not just range over `JVMStats.JVM` (and others), instead of using `p` when adding fields, just use the string `jvm` (or others)? similar to how you do input, filters, and output plugins.